modules/block/src/main/kotlin/jp/co/soramitsu/iroha2/Genesis.kt

```diff
@@ -27,6 +28,17 @@ open class Genesis(open val genesisBlock: RawGenesisBlock) {
     fun asJson(): String = JSON_SERDE.writeValueAsString(this.genesisBlock)
 
     companion object {
+        /**
+         * List of genesis blocks to single block with unique instructions
+         */
+        fun List<Genesis>.toSingle(): Genesis {
+            val uniqueIsi: MutableSet<Instruction> = mutableSetOf()
+            this.forEach { genesis ->
+                uniqueIsi.addAll(genesis.genesisBlock.transactions.map { it.isi }.flatten())
```

Comment:
I guess it's safer to merge transactions (to have the genesis with many transactions), not instructions of all transactions to a single transaction. I'm not sure if there are instructions that assume a transaction creator to be an initiator (like transfer asset from me implicitly or smth like that). If there is no way to do so we can leave this implementation but I still guess transactions are the target unit.